### ISSUE TITLE
Problem on Loss Function

### ISSUE THREAD

**Issue by ZJG0** [facebookresearch/CrypTen#278]
![image](https://user-images.githubusercontent.com/24633447/124135550-7c585980-dab6-11eb-9204-d2e04504c1a9.png)
When I used the loss functon `CrossEntropyLoss`, there is an error shows 

> input and target must have the same size

so, I found the source code. This function requests two inputs `X: tensor` and `Y: tensor`, the size is `X` is [minibatch, num_classes],  `Y`'s size is [minibatch] which is 1D tensor.However, the source code `assert x.size() == y.size(), "input and target must have the same size"` requests `X.size() == Y.size()`. Is it a bug? How can i solve this problem?

**Comment by knottb** [comment]
Hi, thanks for pointing this out.

So this is not actually a bug, but is quite unclear and should be changed on our end. Essentially, the issue is the following:

In order for CrypTen to properly compute the CrossEntropyLoss() function, we need to convert the labels to be one_hot (i.e. a label with target value 3 out of 5 labels should be represented as `torch.tensor([0, 0, 0, 1, 0])`). This can be performed on the encrypted tensor, but is somewhat more efficient to perform prior to encryption, so we never got around to implementing the adjustment inside of CrypTen, instead opting for the user to input a one-hot label vector. We will plan to add support for detecting which encoding the label input has.

In the meantime, there is luckily a very simple fix you can add on your end. The following should be sufficient:

```
P = crypten.cryptensor(preds)
targets_one_hot = torch.nn.functional.one_hot(targets, num_classes)
T = crypten.cryptensor(targets_one_hot)

loss_value = crypten.nn.CrossEntropyLoss()(P, T)
```

Note that `torch.nn.functional.one_hot(targets, num_labels)` will change the shape of the `targets` input from [minibatch] to `[minibatch, num_classes]`, which is the size of the input `x` in the source code above.

**Comment by ZJG0** [comment]
Thanks for your answer, i will try it.
